plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineSdkClasspathContainer.java

```diff
@@ -36,12 +32,12 @@ public String getDescription() {
 
   @Override
   public IClasspathEntry[] getClasspathEntries() {
-    if (cloudSdkLocation != null) {
-      File jarFile = cloudSdkLocation.toPath().resolve(SDK_JAR).toFile();
-      if (jarFile.exists()) {
-        String appEngineToolsApiJar = jarFile.getPath();
+    CloudSdk cloudSdk = new CloudSdkProvider().getCloudSdk();
+    if (cloudSdk.getJavaAppEngineSdkPath() != null) {
```

Comment:
This is weird. You're checking to see if getJavaAppEngineSdkPath is null but then calling a different method.